dali/pipeline/executor/executor.cc

```diff
@@ -348,30 +348,40 @@ void Executor<WorkspacePolicy, QueuePolicy>::RunHelper(OpNode &op_node, Workspac
     if (had_empty_layout) empty_layout_in_idxs.push_back(i);
   }
 
-  if (op.Setup(output_desc, ws)) {
-    DALI_ENFORCE(
-        static_cast<size_t>(ws.NumOutput()) == output_desc.size(),
-        "Operator::Setup returned shape and type information for mismatched number of outputs");
-    DALI_ENFORCE(op.CanInferOutputs(),
-                 "Operator::Setup returned true indicating that it successfully calculated shape "
-                 "and type information for Operator outputs. In that case CanInferOutputs should "
-                 "always return true.");
-    for (int i = 0; i < ws.NumOutput(); i++) {
-      auto &desc = output_desc[i];
-      if (ws.template OutputIsType<CPUBackend>(i)) {
-        ws.template Output<CPUBackend>(i).Resize(desc.shape, desc.type);
-      } else {
-        ws.template Output<GPUBackend>(i).Resize(desc.shape, desc.type);
+  bool ret = false;
+  {
+    DomainTimeRange tr("[DALI][Executor] Setup");
+    ret = op.Setup(output_desc, ws);
+  }
+  {
+    DomainTimeRange tr("[DALI][Executor] Allocate outputs");
```

Comment:
I think this might be a bit misleading. The ops that don't use setup and don't allocate at all would probably report a very short time here, which works ok, but so would something that allocates, but it doesn't use Setup.
The second case is the misleading one. It might raise question why some of the allocations are that fast and if we can do it in other places. But maybe I am overthinking this.

Reply:
I think the main point is to highlight places where the setup and allocation take too much time. If it is fast (because is just fast or no-op) it means we don't have to look into these parts of the operator execution.

Reply:
What I mean by misleading is the operators that don't use setup just allocate within run, so this is the misleading part. But as I said, I'm probably overthinking.